app/models/admin/status_batch_action.rb

```diff
@@ -65,6 +67,38 @@ def handle_delete!
     RemovalWorker.push_bulk(status_ids) { |status_id| [status_id, { 'preserve' => target_account.local?, 'immediate' => !target_account.local? }] }
   end
 
+  def handle_mark_as_sensitive!
+    # Can't use a transaction here because UpdateStatusService queues
+    # Sidekiq jobs
+    statuses.includes(:media_attachments).find_each do |status|
+      next unless status.with_media?
```

Comment:
The sensitive flag can also affect preview cards, so I would do without this check: even if it's an edge case, it can be a useful one.